lib/rules/selector-max-compound-selectors/index.mjs

```diff
@@ -64,34 +65,37 @@ const rule = (primary, secondaryOptions) => {
 		 * @param {import('postcss').Rule} ruleNode
 		 */
 		function checkSelector(selectorNode, ruleNode) {
-			let compoundCount = 1;
+			/** @type {import('postcss-selector-parser').Node[]} */
+			const filteredChildNodes = [];
 
-			selectorNode.each((childNode, index) => {
+			selectorNode.each((childNode) => {
 				// Only traverse inside actual selectors and context functional pseudo-classes
-				if (childNode.type === 'selector' || isContextFunctionalPseudoClass(childNode)) {
+				if (isSelector(childNode) || isContextFunctionalPseudoClass(childNode)) {
 					checkSelector(childNode, ruleNode);
 				}
 
-				// Compound selectors are separated by combinators, so increase count when meeting one
-				if (childNode.type === 'combinator') {
-					compoundCount++;
-
-					return;
+				if (!isSelectorIgnored(childNode)) {
+					filteredChildNodes.push(childNode);
 				}
+			});
 
-				// Try ignoring the selector if the current node is the first, or the previous node is a combinator
-				const previousNode = selectorNode.at(index - 1);
+			// Normalize selector nodes and count combinators
+			const combinatorCount = filteredChildNodes.reduce((count, node, i, nodes) => {
+				// Not count a node except a combinator
+				if (!isCombinator(node)) return count;
 
-				if ((!previousNode || previousNode.type === 'combinator') && isSelectorIgnored(childNode)) {
-					compoundCount--;
-				}
-			});
+				// Not count a combinator at the edge
+				if (i === 0 || i === nodes.length - 1) return count;
+
+				// Not count a consecutive combinator
+				if (isCombinator(nodes[i + 1])) return count;
+
+				return count + 1;
+			}, 0);
+
+			const compoundCount = combinatorCount + 1;
 
-			if (
-				selectorNode.type !== 'root' &&
-				selectorNode.type !== 'pseudo' &&
-				compoundCount > primary
-			) {
+			if (!isRoot(selectorNode) && !isPseudo(selectorNode) && compoundCount > primary) {
```

Comment:
minor
perf

You can put an early `if (isRoot(selectorNode) || isPseudo(selectorNode)) return;` in between l80 and l83 so that we don't call the `reduce` in those cases; ending up with

```js
if (compoundCount > primary) {
```

Reply:
Nice catch! Fixed via debfdec55 👍🏼